benchmarks/dbscan/dbscan_timpl.hpp

```diff
@@ -263,10 +263,8 @@ bool ArborXBenchmark::run(ArborXBenchmark::Parameters const &params)
   using Primitives = decltype(primitives);
 
   using ArborX::DBSCAN::Implementation;
-  Implementation implementation;
-  if (params.implementation == "fdbscan")
-    implementation = Implementation::FDBSCAN;
-  else if (params.implementation == "fdbscan-densebox")
+  Implementation implementation = Implementation::FDBSCAN;
+  if (params.implementation == "fdbscan-densebox")
```

Comment:
This is bug prone you should not have done that.
Where are the string values validated?

Reply:
Validation happens here
https://github.com/arborx/ArborX/blob/c3fd09135eb4ec79369df53fd91218b4ec69fa62/benchmarks/dbscan/dbscan.cpp#L93-L98
but I still think this change is a bad idea

Reply:
The validation is done in the driver, which is responsible for checking that it is one of the "fdbscan" or "fdbscan-densebox". I could have duplicated it here, but it is not strictly necessary.

Reply:
My point is that tools are pretty good at spotting use of uninitialized values.  This change makes it harder to track down potential future bugs.

Reply:
I'm not sure about your point. I was getting warnings when building. I merged the PR with HIP failing, which means that I missed the warning.

I can revert the commit and fix it by providing an extra `else` branch, if so desired.